test/unit/core/viz/expressions/ramp.test.js

```diff
@@ -17,19 +20,374 @@ describe('src/core/viz/expressions/ramp', () => {
-            expect(r2.eval()).toEqual(57);
+                it('should get the first value', () => {
+                    const r = ramp(0, values);
+
+                    r._compile();
+                    actual = r.eval();
+                    expected = values[0];
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should be able to get the second value', () => {
+                    const r = ramp(1, values);
+
+                    r._compile();
+                    actual = r.eval();
+                    expected = values[1];
+
+                    expect(actual).toEqual(expected);
+                });
+            });
+
+            describe('and values are colors', () => {
+                const firstColor = s.namedColor('red');
+                const secondColor = s.namedColor('blue');
+                let actual;
+                let expected;
+
+                it('should get the first value', () => {
+                    const r = ramp(0, [firstColor, secondColor]);
+
+                    r._compile();
+                    actual = r.eval();
+                    expected = firstColor._nameToRGBA();
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should get the second value', () => {
+                    const r = ramp(1, [firstColor, secondColor]);
+
+                    r._compile();
+                    actual = r.eval();
+                    expected = secondColor._nameToRGBA();
+
+                    expect(actual).toEqual(expected);
+                });
+            });
+        });
+    });
+
+    describe('.eval with buckets', () => {
+        describe('when palettes are color arrays', () => {
+            const METADATA = {
+                columns: [
+                    {
+                        name: 'grade',
+                        type: 'category',
+                        categoryNames: [
+                            'A',
+                            'B',
+                            'C',
+                            'D',
+                            'E'
+                        ],
+                    },
+                ],
+                categoryIDs: {
+                    'A': 'A',
+                    'B': 'B',
+                    'C': 'C',
+                    'D': 'D',
+                    'E': 'E'
+                }
+            };
+            
+            const red = s.namedColor('red');
+            const blue = s.namedColor('blue');
+            const yellow = s.namedColor('yellow');
+            const purple = s.namedColor('purple');
+            const green = s.namedColor('green');
+            const orange = s.namedColor('orange');
+
+            let actual;
+            let expected;
+
+            describe('and there are less categories than colors', () => {
+                describe('with "others"', () => {
+                    it('should not show interpolation', () => {
```

Comment:
`should return a non-interpolated color`

Reply:
ok, I just found all colors are interpolated so we need a better description here